src/compiler/Restler.Compiler/Annotations.fs

```diff
@@ -220,4 +221,70 @@ let getGlobalAnnotationsFromFile filePath =
             printfn "ERROR: invalid annotation file: x-restler-global-annotations must be the key"
             raise (ArgumentException("invalid annotation file"))
     else
-        List.empty
\ No newline at end of file
+        List.empty
+
+// Gets the REST-ler dependency annotation from OpenAPI v3 links
+//    "links": {
+//      "linkName":{
+//        "description": "description",
+//        "operationId": "target operation",
+//        "parameters": {
+//          "configStoreName": "JSON path to value"
+//        }
+//      }
+//    }
+//
+let getAnnotationsFromOpenapiLinks (producerRequestId:RequestId) (links:IDictionary<_, NSwag.OpenApiLink>) (oasDoc:NSwag.OpenApiDocument) =
+
+    // Parse the producer parameter from the JSON path in the parameter value
+    // Only match specific patterns for now
+    let parseProducerParameter (paramValue:obj) =
+        // OAS v3 says that the value can be "any" but we only support strings
+        match paramValue with
+        | :? string as s ->
+            match s.Split(".") with
+            | [|"$request"; "path" | "query" | "header"; v|] -> v |> ResourceName |> Some
+            | [|"$response"; "header"; v|] -> v |> ResourceName |> Some
+            | [|"$response"; v|] when v.StartsWith("#body") -> v.Replace("#body","") |> ResourceName |> Some
+            | _ -> None
+        | _ -> None
+
+    let parseLink (link:NSwag.OpenApiLink): ProducerConsumerAnnotation option =
+
+        match oasDoc.Operations |> Seq.tryFind (fun op -> op.Operation.OperationId = link.OperationId) with
+        | None -> None
+        | Some consumerOperation ->
+            let consumerEndpoint = consumerOperation.Path
+            let consumerMethod = getOperationMethodFromString (consumerOperation.Method.ToString())
+            let consumerRequestId = Some {
+                                        endpoint = consumerEndpoint
+                                        method = consumerMethod
+                                        xMsPath = None
+                                    }
+            // For now we only support a single parameter
+            //let consumerParameter = link.Parameters |> Seq.tryHead |> Option.map (fun (k,v) -> ResourceName k)
+            let foo = link.Parameters |> Seq.tryHead
+            match link.Parameters |> Seq.tryHead with
+            | None ->
+                printfn "ERROR: malformed annotation, no parameters specified for link %s" link.OperationId
+                None
+            | Some kv ->
+                // Note: OAS v3 says
+                // "The parameter name can be qualified using the parameter location [{in}.]{name} for operations
+                // that use the same parameter name in different locations (e.g. path.id)."
+                let consumerParameter = Regex.Replace(kv.Key, "^(path\.|query\.|header\.|cookie\.)", "") |> ResourceName |> Some
```

Comment:
>consumerParameter [](http://example.com/codeflow?start=20&length=17)

can the consumer parameter not be in the body, or is this just not supported right now?

Reply:
It can't -- there is no syntax for this in OpenAPI links. The consumer must be a "parameter". I've added that to the doc in Links.md.